### DISCUSSION TITLE
How can I disable collect model data?

### DISCUSSION THREAD

**Discussion by benfiratkaya** [spatie/laravel-permission#2049] [Q&A]
Hello, I don't want to collect model_type and guard_name. I don't think this is good database design.
I design database for permissions like this:

**roles:**
- id
- name
- slug

**permissions:**
- id
- name
- slug

**role_permission:**
- role_id
- permission_id

**user_roles:**
- user_id
- role_id

**user_permissions:**
- user_id
- permission_id

Can I design laravel-permissions based on this database?

**Comment by erikn69** [comment]
Yes, you can
You have to overwrite laravel relations on migration

**Comment by angeljqv** [comment]
> I don't want to collect model_type and guard_name. I don't think this is good database design.

It's because you don't undestand how laravel guards works

**Reply by benfiratkaya** [reply]
I say this because database designed for laravel app. What happens if I work with node.js app and connection to the database? This design good for only laravel app.

**Reply by angeljqv** [reply]
Laravel is the main requirement
https://github.com/spatie/laravel-permission/blob/b731394245a0b88ed255b62bcad34b1fc163b625/composer.json#L26-L29

**Reply by benfiratkaya** [reply]
Yes I know... The database just doesn't work for laravel. It can serve other applications as well. The design not good for other applications

**Reply by angeljqv** [reply]
> The database just doesn't work for laravel

Of course it works

>The design not good for other applications

this is a laravel package, this section is for discussions of laravel, php, related to this package, other applications do not matter here, only laravel

**Comment by fredsal** [comment]


**Reply by fredsal** [reply]
@benfiratkaya mark this as answered, so no one else wastes time trying to answer it
